frontend/src/metabase/api/settings.ts

```diff
@@ -10,11 +13,13 @@ import { invalidateTags, tag } from "./tags";
 export const settingsApi = Api.injectEndpoints({
   endpoints: (builder) => ({
     // admin-only endpoint that returns all settings with lots of extra metadata
-    getAdminSettingsDetails: builder.query<SettingDefinition[], void>({
+    getAdminSettingsDetails: builder.query<SettingDefinitionMap, void>({
       query: () => ({
         method: "GET",
         url: "/api/setting",
       }),
+      transformResponse: (response: SettingDefinition[]) =>
+        _.indexBy(response, "key") as SettingDefinitionMap,
```

Comment:
This cast actually narrows the type, underscore would only yield 
- Record<EnterpriseSettingKey, SettingDefinition>
But wouldn't reflect the fact that the internals of SettingDefinition are further narrowed by the key